File: Backend/src/main/java/com/example/bus_reservation_system/entity/Reservation.java
package com.example.bus_reservation_system.entity;

import jakarta.persistence.*;
import lombok.Data;

import java.sql.Date;
import java.time.LocalDateTime;

@Entity
@Data
@Table(name = "Reservation")
public class Reservation {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    @Column(name="user_id",nullable=false)
    private long user_id;

    @Column(name="bus_id",nullable=false)
    private long bus_id;

    @Column(nullable = false)
    private String reservationDate;

    @Column(nullable = false)
    private int seatNumber;



    @Column(nullable = false)
    private long bookingId;




}

File: Frontend/src/app/Models/reservation.ts

export class Reservation {
    id?: number; 
    userId: number; 
    bus_id: number; 
    reservationDate: string; 
    seatNumber: number; 
    bookingId:number;
    status:string;
    
    constructor(
       
        userId: number,
        bus_id: number = 0,
        reservationDate: string='' ,
        seatNumber: number = 0,
        bookingId:number=0,
        status:string='',
        id?: number
        
    ) {
        this.id = id;
        this.userId = userId;
        this.bus_id = bus_id;
        this.reservationDate = reservationDate;
        this.seatNumber = seatNumber;
        this.bookingId=bookingId;
        this.status=status;
       
  
    }

    // You can add methods here if needed
}